### PR TITLE
[iOS][Blemulator] connectToDevice

### PR DESCRIPTION
**Pull request by TomoLV**

### CODE REVIEW

File: ios/Classes/Blemulator/Bridging/DartMethodCaller.m [hunk 1]
```diff
@@ -41,6 +47,31 @@ - (void)stopDeviceScan {
...
+    NSDictionary<NSString *,id> *arguments = [NSDictionary dictionaryWithObjectsAndKeys:
+                                               deviceIdentifier, SIMULATION_ARGUMENT_NAME_DEVICE_ID,
+                                               nil];
+    SuccessHandler successHandler = ^(id _Nullable result) {
+        resolve([[[Peripheral alloc] initWithIdentifier:deviceIdentifier
+                                                   name:name
+                                                    mtu:23] jsonObjectRepresentation]);
+    };
+    ErrorHandler errorHandler = ^(id error) {
+        // TODO: - Send error here
```
Comment (mikolak) [inline ios/Classes/Blemulator/Bridging/DartMethodCaller.m]: Create issue for error handling

File: ios/Classes/Blemulator/Bridging/DartMethodCaller.m [hunk 2]
```diff
@@ -55,4 +86,21 @@ - (InvokeMethodResultHandler)simpleInvokeMethodResultHandlerForMethod:(NSString
     };
 }
 
+- (InvokeMethodResultHandler)invokeMethodResultHandlerForMethod:(NSString *)methodName
+                                                        onSuccess:(SuccessHandler)successHandler
+                                                         onError:(ErrorHandler)errorHandler {
+    return ^(id _Nullable result) {
+        if ([result class] == [FlutterError class]) {
+            NSLog(@"%@%@%@%@", @"DartMethodCaller.", methodName, @": FlutterError: ", [(FlutterError *)result message]);
```
Comment (mikolak) [inline ios/Classes/Blemulator/Bridging/DartMethodCaller.m]: What's %@%@%@?

File: ios/Classes/Blemulator/Model/ConnectionStateEvent.m [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    if (self) {
+        self.deviceId = deviceId;
+        self.connectionState = connectionState;
+    }
+    return self;
+}
+
+- (NSDictionary<NSString *,id> *)jsonObjectRepresentation {
+    return [NSDictionary dictionaryWithObjectsAndKeys:
+            _deviceId, @"id",
```
Comment (mikolak) [inline ios/Classes/Blemulator/Model/ConnectionStateEvent.m]: Will those strings be reused? Perhaps they should be extracted?

File: ios/Classes/Blemulator/Bridging/DartMethodCaller.m (hunk 2 above)
Comment (TomoLV) [reply ios/Classes/Blemulator/Bridging/DartMethodCaller.m]: It's for specyfing the format of the arguments in output of NSLog.

File: ios/Classes/Blemulator/Bridging/DartMethodCaller.m (hunk 1 above)
Comment (TomoLV) [reply ios/Classes/Blemulator/Bridging/DartMethodCaller.m]: Issue created

File: ios/Classes/Blemulator/Model/ConnectionStateEvent.m (hunk 3 above)
Comment (TomoLV) [reply ios/Classes/Blemulator/Model/ConnectionStateEvent.m]: I'm not sure yet if they are going to be reused. I've extracted them to local constants for better code structure though - it's possible to make them public if there's a need in the future.

**Pull request merged**